### PR TITLE
GSoC 2019: Supports `comm=none`-only configuration for UnitTest Launcher

### PR DESCRIPTION
**Pull request by krishnadey30**
This PR adds a runtime `CHPL_COMM` check to the launcher, which solves the following issue
https://github.com/chapel-lang/chapel/issues/13590

Also updates the test for `comm=none`

### CODE REVIEW

File: test/library/packages/UnitTest/Launcher.chpl [hunk 1]
```diff
@@ -58,6 +58,23 @@ module Launcher {
         if files.size == 0 && dirs.size == 0 {
           dirs.push_back(".");
         }
+        // set comm if found empty
+        if comm == "" then comm = "none";
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/Launcher.chpl]: I just realized there is an error in our strategy here -- `CHPL_COMM` default value may depend on several factors if `CHPL_COMM` is not explicitly set.

I think we're going to actually need to have our launcher call `$CHPL_HOME/util/printchplenv` or `python $CHPL_HOME/util/chplenv/chpl_comm.py` directly to determine the `CHPL_COMM` value.

This could be wrapped in a clean interface like `proc getRuntimeComm();`, which should eliminate the need for a lot of conditional logic here.

File: test/library/packages/UnitTest/Launcher.chpl [hunk 2]
```diff
@@ -179,7 +195,7 @@ module Launcher {
     if erroredTestNames.size != 0 then erroredTestNamesStr = erroredTestNames: string;
     if testsPassed.size != 0 then passedTestStr = testsPassed: string;
     if skippedTestNames.size != 0 then skippedTestNamesStr = skippedTestNames: string;
-    var exec = spawn(["./"+executable, "-nl", reqNumLocales: string, "--testNames", 
+    var exec = spawn(["./"+executable, "--numLocales", reqNumLocales: string, "--testNames",
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/Launcher.chpl]: What is the impact of this change? or is it just stylistic?

Comment (krishnadey30) [reply test/library/packages/UnitTest/Launcher.chpl]: I thought `-nl` doesn't work when `comm=none`

Comment (krishnadey30) [reply test/library/packages/UnitTest/Launcher.chpl]: Have reverted to previous version.

File: test/library/packages/UnitTest/Launcher.chpl (hunk 1 above)
Comment (krishnadey30) [reply test/library/packages/UnitTest/Launcher.chpl]: okay sir.
I have made the changes.

File: test/library/packages/UnitTest/Launcher.chpl [hunk 3]
```diff
@@ -15,6 +15,32 @@ module Launcher {
   config const setComm: string = "";
   var comm: string;
 
+  /* Gets the comm */
+  proc getRuntimeComm() throws {
+    var line: string;
+    var checkComm = spawn(["python",CHPL_HOME:string+"/util/chplenv/chpl_comm.py"],
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/Launcher.chpl]: I wonder if it will matter if the user is working with a *different* chapel installation than the one that compiled the launcher, i.e. `runtime CHPL_HOME != compile-time CHPL_HOME`.

File: test/library/packages/UnitTest/Launcher.chpl [hunk 4]
```diff
@@ -15,6 +15,32 @@ module Launcher {
...
+                        stdout = PIPE);
+    while checkComm.stdout.readline(line) {
+      comm = line.strip();
+    }
+    // setting communication mechanism.
+    if setComm != "" {
+      if comm != "none" {
```
Comment (ben-albrecht) [inline test/library/packages/UnitTest/Launcher.chpl]: Shouldn't `setComm` override `comm` regardless of the value from `comm` or am I confused about the intended behavior of `setComm`? It seems like this logic could be cleaned up a little.

Comment (ben-albrecht) [reply test/library/packages/UnitTest/Launcher.chpl]: I would add a doc-string to `setComm` to clarify the purpose / rules as well.

Comment (krishnadey30) [reply test/library/packages/UnitTest/Launcher.chpl]: `setComm` will override the `comm`. But if the `comm=none` and user wants to run in `gasnet`, should we allow that?

File: test/library/packages/UnitTest/Launcher.chpl (hunk 3 above)
Comment (krishnadey30) [reply test/library/packages/UnitTest/Launcher.chpl]: No, it doesn't matter where you compile the launcher as the use of `comm` is just to verify that you are in a multilocale mode.

File: test/library/packages/UnitTest/Launcher.chpl (hunk 4 above)
Comment (ben-albrecht) [reply test/library/packages/UnitTest/Launcher.chpl]: Ah I see. No that shouldn't be necessary right now. Thanks.
